perfkitbenchmarker/providers/ibmcloud/ibmcloud_network.py

```diff
@@ -0,0 +1,243 @@
+
+  def _Create(self):
+    """Creates a IBM Cloud VPC with address prefixes for all zones."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['name'] = self.name
+    logging.info('Creating vpc: %s', cmd.flags)
+    self.vpcid = cmd.CreateVpc()
+    if self.vpcid:
+      for zone in FLAGS.zones:
+        cmd.flags.update({
+          'vpcid': self.vpcid,
+          'zone': zone
+          })
+        index = int(zone[len(zone) - 1])  # get the ending -1
+        cmd.flags['cidr'] = VPC_PREFIX_RANGES[index - 1]
+        cmd.flags['name'] = self.prefix + VPC_NAME + util.DELIMITER + zone
+        resp = cmd.CreatePrefix()
+        if resp:
+          logging.info('Created vpc prefix range: %s', resp.get('id'))
+        else:
+          raise errors.Error('IBM Cloud ERROR: Failed to create '
+                             'vpc address prefix')
+    else:
+      raise errors.Error('IBM Cloud ERROR: Failed to create vpc')
+
+  def CreateSubnet(self, vpcid):
+    """Creates a IBM Cloud subnet on the given vpc."""
+    self.vpcid = vpcid
+    subnet_index = int(self.zone[len(self.zone) - 1])  # get the ending -1
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags.update({
+      'vpcid': self.vpcid,
+      'zone': self.zone,
+      'name': self.prefix + VPC_NAME + util.SUBNET_SUFFIX + \
+      str(subnet_index) + util.DELIMITER + self.zone,
+      'cidr': VPC_SUBNETS[subnet_index - 1]
+      })
+    logging.info('Creating subnet: %s', cmd.flags)
+    resp = cmd.CreateSubnet()
+    if resp:
+      self.subnet = resp.get('id')
+      IbmCloudVirtualMachine.validated_subnets += 1
+      logging.info('Created subnet: %s, zone %s', self.subnet, self.zone)
+    else:
+      raise errors.Error('IBM Cloud ERROR: Failed to create subnet')
+
+  def CreateFip(self, name: str, vmid: str):
+    """Creates a VNIC in a IBM Cloud VM instance."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['instanceid'] = vmid
+    vnicid = cmd.InstanceGetPrimaryVnic()
+    cmd.flags['name'] = name
+    cmd.flags['target'] = vnicid
+    logging.info('Creating FIP for instanceid: %s', vmid)
+    resp = json.loads(cmd.InstanceFipCreate())
+    if resp:
+      logging.info('FIP create resp: %s', resp)
+      assert resp['address'] != None
+      return resp['address'], resp['id']
+    logging.error('FAILED to create FIP for instance %s', vmid)
+    return None, None
+
+  def DeleteFip(self, vmid: str, fip_address: str, fipid: str):
+    """Deletes fip in a IBM Cloud VM instance."""
+    logging.info('Deleting FIP, instanceid: %s, fip address: %s, fip id: %s',
+                 vmid, fip_address, fipid)
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['fipid'] = fipid
+    cmd.InstanceFipDelete()
+
+  def _Exists(self):
+    """Returns true if the VPC exists."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags.update({
+      'prefix': self.prefix,
+      'zone': self.zone,
+      'items': 'vpcs'
+      })
+    self.vpcid = cmd.ListResources()
```

Comment:
I don't know the signature of ListResources but it seems like that should return a list of items, not just a single one.  Also is it returning just the resource IDs?

Reply:
I renamed the function name to GetResource(), that was confusing as it returns only first id that matches the prefix.